src/erlcloud_s3.erl

```diff
@@ -1227,6 +1232,228 @@ set_bucket_attribute(BucketName, AttributeName, Value, Config)
+    when is_list(BucketName), is_list(Inventory) ->
+
+    put_bucket_inventory(BucketName, default_config()).
+
+-spec put_bucket_inventory(string(), list(), aws_config())
+        -> ok | {error, Reason::term()}.
+put_bucket_inventory(BucketName, Inventory, #aws_config{} = Config)
+    when is_list(BucketName), is_list(Inventory) ->
+
+    InventoryId = proplists:get_value(id, Inventory),
+    XmlInventory = encode_inventory(Inventory),
+    put_bucket_inventory(BucketName, InventoryId, list_to_binary(XmlInventory), Config).
+
+-spec put_bucket_inventory(string(), string(), binary(), aws_config())
+        -> ok | {error, Reason::term()}.
+put_bucket_inventory(BucketName, InventoryId, XmlInventory, #aws_config{} = Config)
+    when is_list(BucketName), is_list(InventoryId), is_binary(XmlInventory) ->
+    Md5 = base64:encode(crypto:hash(md5, XmlInventory)),
+    Params = [{"id", InventoryId}],
+    Headers = [{"Content-MD5", Md5}, {"content-type", "application/xml"}],
+    s3_simple_request(Config, put, BucketName, "/", "inventory", Params, XmlInventory, Headers).
+
+-spec delete_bucket_inventory(string(), string()) -> ok | {error, Reason::term()}.
+delete_bucket_inventory(BucketName, InventoryId) when is_list(BucketName), is_list(InventoryId) ->
+    delete_bucket_inventory(BucketName, InventoryId, default_config()).
+
+-spec delete_bucket_inventory(string(), string(), aws_config()) -> ok | {error, Reason::term()}.
+delete_bucket_inventory(BucketName, InventoryId, #aws_config{} = Config)
+    when is_list(BucketName), is_list(InventoryId) ->
+
+    Params = [{"id", InventoryId}],
+    s3_simple_request(Config, delete, BucketName, "/", "inventory", Params, <<>>, []).
+
+encode_inventory(Inventory) ->
+    lists:flatten(
+        xmerl:export_simple(
+            [{
+                'InventoryConfiguration',
+                [{xmlns, "http://s3.amazonaws.com/doc/2006-03-01/"}],
+                inv_encode_subtype(Inventory)
+            }],
+            xmerl_xml
+        )
+    ).
+
+inv_encode_subtype([{_, _} | _] = Proplist) ->
+    lists:flatten(lists:map(
+        fun(Elem) ->
+            inv_encode_subtype(Elem)
+        end,
+        Proplist
+    ));
+inv_encode_subtype({Key, [{_, _} | _] = Proplist}) ->
+    {inv_key_to_name(Key), inv_encode_subtype(Proplist)};
+inv_encode_subtype({Key, [[{_, _} | _], _] = ListOfProplist}) ->
+    lists:map(
+        fun(List) ->
+            {inv_key_to_name(Key), inv_encode_subtype(List)}
+        end,
+        ListOfProplist
+    );
+inv_encode_subtype({Key, [Val | _] = ListOfValues}) when is_list(Val) ->
+    lists:map(
+        fun(Value) ->
+            {inv_key_to_name(Key), inv_encode_subtype(Value)}
+        end,
+        ListOfValues
+    );
+inv_encode_subtype({Key, Value}) ->
+    {inv_key_to_name(Key), inv_encode_subtype(Value)};
+inv_encode_subtype(Value) ->
+    [erlcloud_util:to_string(Value)].
+
+inv_key_to_name(inventory_configuration) ->  'InventoryConfiguration';
```

Comment:
`' '`
atoms like `inventory_configuration` are more usable in a code while `'InventoryConfiguration'` are required by xml parser to build xml